src/main/java/ru/marina/tshop/users/UserService.java

```diff
@@ -0,0 +1,68 @@
+package ru.marina.tshop.users;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ru.marina.tshop.security.JwtTokenProvider;
+import ru.marina.tshop.utils.IdGenerator;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+public class UserService {
+    private final IdGenerator idGenerator;
+    private final UserDao userDao;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtTokenProvider jwtTokenProvider;
+
+    @Autowired
+    public UserService(final IdGenerator idGenerator, final UserDao userDao,
+                       final PasswordEncoder passwordEncoder, final JwtTokenProvider jwtTokenProvider) {
+        this.idGenerator = idGenerator;
+        this.userDao = userDao;
+        this.passwordEncoder = passwordEncoder;
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Transactional
+    public String register(final String email,
+                           final String firstName,
+                           final String lastName,
+                           final LocalDate birthDate,
+                           final String phone,
+                           final String password,
+                           final List<Role> roles) {
+        final String userId = idGenerator.generateId();
+
+        if (userDao.findByEmail(email).isPresent()) {
+            throw new RuntimeException("User with this login is already exist");
```

Comment:
There should be a dedicated exception like `EmailAlreadyUsedException` thrown to indicate a particular issue.